### PR TITLE
ClockworkPi's GameShell port

### PR DESCRIPTION
**Pull request by pleft**
Added `Makefile.gameshell` to compile against ClockworkPi's GameShell arm-based game console (https://www.clockworkpi.com/). Also added a default key mapping which corresponds to GameShell's gamepad layout. Compilation is supposed to take place on the GameShell itself (no cross-compiling).

A successful build has already been released https://forum.clockworkpi.com/t/lets-play-jazz-jackrabbit/1470

### CODE REVIEW

File: src/io/controls.cpp [hunk 1]
```diff
@@ -137,7 +148,7 @@
     #define DEFAULT_BUTTON_STATS           (10)
     #define DEFAULT_BUTTON_PAUSE           (11)
     #define DEFAULT_BUTTON_YES             (-1)
-    #define DEFAULT_BUTTON_NO              (-1)
+    #define DEFAULT_BUTTON_NO              (-1)
```
Comment (carstene1ns) [inline src/io/controls.cpp]: Typo?

**Comment by carstene1ns** [comment]
Looks good to me so far.
I have removed most of the old `Makefile`s in my [buildsystem branch](https://github.com/AlisterT/openjazz/compare/buildsystem), can we use the autotools on this device? Would be nice to not have added another `Makefile`. :+1: At least does not seem to contain anything special for this platform.

File: src/io/controls.cpp (hunk 1 above)
Comment (pleft) [reply src/io/controls.cpp]: Line `#define DEFAULT_BUTTON_NO              (-1)	` must be a typo, accidentally introduced a `tab`.

**Comment by pleft** [comment]
Hi, I guess we can use autotools. Do you want me to provide the `host` parameter of the platform?

**Comment by carstene1ns** [comment]
That would be awesome, i figured it would be `arm-linux-gnueabihf`?

Button configuration looks ok, does this device not provide a joystick?

**Comment by pleft** [comment]
I will check for the host parameter tonight, as for the buttons, this device's gamepad is recognized as keyboard so for example the START button maps to the ENTER key.

**Comment by pleft** [comment]
Here is the host info:

```bash
cpi@clockworkpi:/usr/share/automake-1.15$ ./config.guess
armv7l-unknown-linux-gnueabihf
```

**Comment by carstene1ns** [comment]
Great, now only a bit of polishing left... ;-)

**Comment by pleft** [comment]
done please review. :)

**Pull request merged**